### PR TITLE
Allow for customised content-type validation

### LINKED ISSUE #80482: Allow RestHandlers for Content-Type outside XContentType
It is currently impossible for `RestHandler` to deal with requests that specify a `Content-Type` that doesn't map to `XCotentType` due to strict validation inside `RestController.`
This makes dealing with binary data (for supporting endpoints/protocols that already exist) impossible since the handler never gets access to the underlying request - the sender receiving the response: `Content-Type header [<non-xcontent-type>] is not supported`

It would be great to allow the validation to be pluggable similar to the `Allow` header - a `RestHandler` could return its own `MediaTypeRegistry` to specify the accepted `Content-Type` (including requests without the header being defined).

### PR DESCRIPTION
**Pull request by pgomulka**
In order to support additional media types in request body a custom
validation has to be supported.
This commit moves validation from RestController to RestHandler
interface (default) and allows new RestHandler implementations to
provide its custom implementation

closes #80482

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by pgomulka** [comment]
@elasticmachine update branch

**Comment by pgomulka** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/rest/RestHandler.java [hunk 1]
```diff
@@ -76,8 +73,8 @@ default boolean allowSystemIndexAccessByDefault() {
         return false;
     }
 
-    default MediaTypeRegistry<? extends MediaType> validAcceptMediaTypes() {
-        return XContentType.MEDIA_TYPE_REGISTRY;
```
Comment (grcevski) [inline server/src/main/java/org/elasticsearch/rest/RestHandler.java:80]: Just for my own learning, was nobody using this interface method before? Do we need to still report somewhere that the default supported media type is MEDIA_TYPE_REGISTRY?

**Review by grcevski — APPROVED** [review]
LGTM!

**Comment by tvernum** [comment]
@pgomulka On the original issue we discussed maintaining CSRF protection by continuing to reject any of the safelisted content types.

It doesn't look like this PR does that. Was that intentional?

**Comment by pgomulka** [comment]
great find @tvernum . It wasn't intentional, it was a while when I read the discussion and assumed the PR was ready.
I missed that the explicit check for XContentType values on Content-Type header was preventing the safelisted to be sent.. I will follow up in a new PR to fix this.
thanks again!

**Pull request merged**